### DISCUSSION TITLE
Add some checking for blank base_path setting ...

### DISCUSSION THREAD

**Discussion by jcogs-design** [ExpressionEngine/ExpressionEngine#2878] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
EE6.4.0 - but apparently true also for EE7... 

The Control Panel setting for Base Path does not apply any validation check if the entry is blank (for some reason it is not a required parameter).

Related to this - if the base_path value is blank and entry for "Themes Path" is set to `{base_path}themes/` EE does not attempt to validate the Themes Path as valid (even though it clearly would not be in this scenario). Presumably because it has some content ... 

If you enter junk into the base_path entry then EE validates both the base_path setting *and* the Themes Path setting and identifies both as invalid.

Experience (from JCOGS Image user bug reports) suggests that sites updating from EE2 sometimes arrive at EE6 or EE7 with the base_path value empty.

A blank base path value possibly does not cause any issues for EE itself, but does cause problems for any add-ons that attempt to write to folders within the web root of the EE site - since they are unable to work out where the web root is. Presumably also users with blank base_path settings and Theme Path set to `{base_path}themes/` will loose CP styling too.

**Describe the solution you'd like**
Add a check to the URL and Path Settings page in CP to ensure that EE does not allow a 'blank' base_path value (e.g. by making this value a 'required' one).
Add a check to the EE2->EE7 auto-update process to identify cases where the base_path setting is empty and flag this up in CP when update is complete (much like it does for template tag issues etc.)

**Describe alternatives you've considered**
Adding even more error trapping to JCOGS Image - the EE call `ee('Filesystem')->getDirectoryContents($path)` unfortunately responds to being given an invalid path (e.g. anything appended to a blank value for base_path) with a php Exception error which doesn't go down well with add-on users... :( 

**Teachability, Documentation, Adoption, Migration Strategy**
As the Theme Path entry shows, Users are comfortable with having 'required' fields in the URL and Path Settings panel in CP, so adding one more would not be a big deal I think.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@bryannielsen you probably have the most knowledge here.

What do you think about solutions proposed? (I don't really like the idea of stopping people from updating, but maybe we could do this as part of post-update checks?)
I believe also empty `{base_path}` is sometimes causing issues with 1-click updates

**Reply by intoeetive** [reply]
https://github.com/ExpressionEngine/ExpressionEngine/pull/3920
